### DISCUSSION TITLE
onRender issues on latest grafana update

### DISCUSSION THREAD

**Discussion by qxlwn** [gapitio/gapit-htmlgraphics-panel#122] [Q&A]
Hi,

Having a bit strange issue with HTML graph since latest grafana update. Grafana v9.1.4

`const htmlgraphicsValue = htmlNode.getElementById('htmlgraphics-value');

if (htmlgraphicsValue) {
  var value = "";
var i = 0;
var text = {};
while (i < data.series.length){
  const valueField = data.series[i]?.fields[1];
  if (valueField) {
    const length = valueField.values.length;
   var result = valueField.values.get(length - 1)`

this has to be adjusted somehow to make it work again. But for now if it stays like it always has been then I get 
![image](https://user-images.githubusercontent.com/113666747/190578091-e2be761b-eaca-4000-b925-58ccef41d125.png)

\\\\\\\\\\\\\\\\\\\\\\\\

I have adjusted a bit so it looks like this and at least it shows data, but still it shows just some small part of it

`const htmlgraphicsValue = htmlNode.getElementById('htmlgraphics-value');

if (htmlgraphicsValue) {
  var value = "";
var i = 0;
var text = {};
while (i < data.series[0].fields[2].values.length){
  const valueField = data.series[0].fields[2];
  if (valueField) {
    const length = valueField.values.length;
   var result = valueField.values.get(length-1)`

**Comment by ZuperZee** [comment]
Most likely the fields you're reading are not present or have changed.
I would suspect the field to be field[1] instead of field[2]. As field[1] is usually where the value field is.

Could you try sending a bit more info about what your data looks like?

**Reply by qxlwn** [reply]
If field is [1] , then it gives out error, but if I put field [2] , then it at least gives out result, but it gives only one result repeatedly. I need to make to give not only one result but all results. 
Also I can make it show different result by changing length -1 to length - 2 for example, but still the result will be given out repeatedly, but as I said already - I need all results not just one.

**Reply by ZuperZee** [reply]
I'm having difficulties recreating your issue.
Could you provide some information on how the data looks?

**Reply by ZuperZee** [reply]
Here's a typical data object logged in the console  `console.log(data)`
![image](https://user-images.githubusercontent.com/39485579/190951212-7bc7efda-240d-4a7e-853c-a8dd0d1b3281.png)
